test/unit/core/utils.ts

```diff
@@ -86,4 +92,24 @@ suite('utils', ({ expect, spy, stub }) => {
       expect(utils.WINDOW()).to.eq(win);
     });
   });
+
+  describe('arrayToDotNotation()', () => {
+    it('should return -1 on invalid input', () => {
+      expect(utils.arrayToDotNotation(<any>1)).to.eq(-1);
+    });
+
+    it('should throw error with throw error flag on invalid input', () => {
```

Comment:
better named as `should throw error on invalid input` or perhaps `... on non-string input`